src/libraries/System.Net.Ping/src/System/Net/NetworkInformation/Ping.cs

```diff
@@ -408,16 +435,28 @@ private PingReply GetAddressAndSend(string hostNameOrAddress, int timeout, byte[
             }
         }
 
-        private async Task<PingReply> GetAddressAndSendAsync(string hostNameOrAddress, int timeout, byte[] buffer, PingOptions? options)
+        private async Task<PingReply> SendPingAsyncInternal<TArg>(
+            TArg getAddressArg,
+            Func<TArg, CancellationToken, ValueTask<IPAddress>> getAddress,
```

Comment:
I did this to reduce logic duplication between the `IPAddress` and `string` paths, but let me know if I should unwind it to keep things how they were before.